vcf-dumper/vcf-dumper-lib/src/test/java/uk/ac/ebi/eva/vcfdump/rules/TestDBRule.java

```diff
@@ -0,0 +1,156 @@
+    public static final String SHEEP_FILE_2_ID = "ERZ324596";
+
+    public static final int NUMBER_OF_SAMPLES_IN_SHEEP_FILES = 453;
+
+    private static final Logger logger = LoggerFactory.getLogger(TestDBRule.class);
+
+    private static final Map<String, String> databaseMapping = new HashMap<>();;
+
+    private MongoClient mongoClient;
+
+
+    @Override
+    protected void after() {
+        cleanDBs();
+        mongoClient.close();
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        mongoClient = new MongoClient();
+        restoreDumpInTemporaryDatabase(HUMAN_TEST_DB_NAME);
+        restoreDumpInTemporaryDatabase(COW_TEST_DB_NAME);
+        restoreDumpInTemporaryDatabase(SHEEP_TEST_DB_NAME);
+    }
+
+    public static String getTemporaryDBName(String databaseName) {
+        return databaseMapping.get(databaseName);
+    }
+
+    private String createRandomTemporaryDatabaseName() {
+        return getTemporaryDatabase().getName();
+    }
+
+    /**
+     * Returns a new temporary database
+     *
+     * @return
+     */
+    private DB getTemporaryDatabase() {
+        return getTemporaryDatabase(UUID.randomUUID().toString());
+    }
+
+    /**
+     * Returns a temporary database with {@param database} name
+     *
+     * @param databaseName
+     * @return
+     */
+    public DB getTemporaryDatabase(String databaseName) {
+        return mongoClient.getDB(databaseName);
+    }
+
+    private void cleanDBs() {
+        for (String databaseName : databaseMapping.values()) {
+            DB database = mongoClient.getDB(databaseName);
+            database.dropDatabase();
+        }
+        databaseMapping.clear();
+    }
+
+    public String restoreDumpInTemporaryDatabase(String database) throws IOException, InterruptedException {
+        URL testDumpDirectory = this.getClass().getResource("/dump/" + database);
+        logger.info("restoring DB from " + testDumpDirectory);
+        String randomDatabaseName = createRandomTemporaryDatabaseName();
+        databaseMapping.put(database, randomDatabaseName);
+        restoreDump(testDumpDirectory, randomDatabaseName);
+        return randomDatabaseName;
+    }
+
+    public void restoreDump(URL dumpLocation, String databaseName) throws IOException, InterruptedException {
+        assert (dumpLocation != null);
+        assert (databaseName != null && !databaseName.isEmpty());
+        String file = dumpLocation.getFile();
+        assert (file != null && !file.isEmpty());
+        getTemporaryDatabase(databaseName);
+
+        logger.info("restoring DB from " + file + " into database " + databaseName);
+
+        Process exec = Runtime.getRuntime().exec(String.format("mongorestore --db %s %s", databaseName, file));
+        exec.waitFor();
+        String line;
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(exec.getInputStream()));
+        while ((line = bufferedReader.readLine()) != null) {
```

Comment:
BufferedReader includes a "lines()" method in Java 8 that returns a Stream<String> object. This loop could be substituted by:
        bufferedReader.lines().forEach(line -> logger.info("mongorestore output: " + line));